source/backend/api/Services/PropertyService.cs

```diff
@@ -296,6 +304,21 @@ public PimsPropertyActivity GetActivity(long propertyId, long activityId)
             throw new BadRequestException("Activity with the given id does not match the property id");
         }
 
+        public PimsPropertyActivity GetFileActivity(long managementFileId, long activityId)
+        {
+            _logger.LogInformation("Retrieving single property Activity...");
+            _user.ThrowIfNotAuthorized(Permissions.ManagementView, Permissions.PropertyView);
+
+            var propertyActivity = _propertyActivityRepository.GetActivity(activityId);
+
+            if (propertyActivity.ManagementFileId == managementFileId)
```

Comment:
Given that the managementFileId is only used to validate the request, let's remove it from this method and move the check to the controller.

Reply:
I thought we agreed to move this logic to the controller and make the service "trust" that the supplied entities were valid....